### ISSUE TITLE
[Bug]: [SDL2) SDL_DestroyWindow - Memory Leak ?

### ISSUE THREAD

**Issue by BeWorld2018** [OpenTTD/OpenTTD#12040]
### Version of OpenTTD

14.0 beta 3

### Expected result

I dont find any SDL_DestroyWindow before SDL_Quit() ? 

https://github.com/OpenTTD/OpenTTD/blob/master/src/video/sdl2_v.cpp#L567

Maybe implement something like that before SDL_QuitSubSystem(SDL_INIT_VIDEO);

```
	if(this->sdl_window)
		SDL_DestroyWindow(this->sdl_window);
```

Thanks

### Actual result

memory leak when exit game ?

### Steps to reproduce

Juste open and close game.

**Comment by TrueBrain** [comment]
It is always good to test a hypothetical before raising it :)

There are several answer to the question, but no, there is no memory leak.

First of, when closing down an application, the only reason `SQL_Quit` is called, "memory leaks" are not really in order. Anything that isn't free'd, will be deallocated anyway, so nothing is leaking memory. When the application is closed, all memory is released.

Second, `SQL_Quit` takes care of all of that. It shuts down all subsystems, and in case of the video driver, it will destroy all windows too [1](https://github.com/libsdl-org/SDL/blob/main/src/video/SDL_video.c#L3778) via [2](https://github.com/libsdl-org/SDL/blob/SDL2/src/SDL.c#L440).

So in short, no, there is no memory leak in either way you slice it. Hope this clears things up :)

**Comment by TrueBrain** [comment]
No worries; feel free to drop by on Discord if you have any more of these questions, or just make a discussion on GitHub :)
